platform/posix/transport/include/openssl_posix.h

```diff
@@ -0,0 +1,189 @@
+
+/* Transport includes. */
+#include "transport_interface.h"
+#include "sockets_posix.h"
+
+/**
+ * @brief Definition of the network context for OpenSSL.
+ *
+ * @note For this transport implementation, the socket descriptor and
+ * SSL context is used.
+ */
+struct NetworkContext
+{
+    int socketDescriptor;
+    SSL * pSslContext;
+};
+
+/**
+ * @brief OpenSSL Connect / Disconnect return status.
+ */
+typedef enum OpensslStatus
+{
+    OPENSSL_SUCCESS = 0,         /**< Function successfully completed. */
+    OPENSSL_INVALID_PARAMETER,   /**< At least one parameter was invalid. */
+    OPENSSL_INSUFFICIENT_MEMORY, /**< Insufficient memory required to establish connection. */
+    OPENSSL_INVALID_CREDENTIALS, /**< Provided credentials were invalid. */
+    OPENSSL_HANDSHAKE_FAILED,    /**< Performing TLS handshake with server failed. */
+    OPENSSL_API_ERROR,           /**< A call to a system API resulted in an internal error. */
+    OPENSSL_DNS_FAILURE,         /**< Resolving hostname of server failed. */
+    OPENSSL_CONNECT_FAILURE      /**< Initial connection to the server failed. */
+} OpensslStatus_t;
+
+/**
+ * @brief Contains the credentials to establish a TLS connection.
+ */
+typedef struct OpensslCredentials
+{
+    /**
+     * @brief An array of ALPN protocols. Set to NULL to disable ALPN.
+     *
+     * See [this link]
+     * (https://aws.amazon.com/blogs/iot/mqtt-with-tls-client-authentication-on-port-443-why-it-is-useful-and-how-it-works/)
+     * for more information.
+     */
+    const char * pAlpnProtos;
+
+    /**
+     * @brief Length of the ALPN protocols array.
+     */
+    size_t alpnProtosLen;
+
+    /**
+     * @brief Set a host name to enable SNI. Set to NULL to disable SNI.
+     */
+    const char * sniHostName;
+
+    /**
+     * @brief Length of the SNI host name.
+     */
+    size_t sniHostNameLen;
+
+    /**
+     * @brief Set this to a non-zero value to use TLS max fragment length
+     * negotiation (TLS MFLN).
+     *
+     * @note The network stack may have a minimum value for this parameter and
+     * may return an error if this parameter is too small.
+     */
+    size_t maxFragmentLength;
+
+    const char * pRootCaPath;     /**< @brief Filepath string to the trusted server root CA. */
+    size_t rootCaPathLen;         /**< @brief Length associated with #NetworkCredentials.pRootCa. */
+    const char * pClientCertPath; /**< @brief Filepath string to the client certificate. */
+    size_t clientCertPathLen;     /**< @brief Length associated with #NetworkCredentials.pClientCert. */
+    const char * pPrivateKeyPath; /**< @brief Filepath string to the client certificate's private key. */
+    size_t privateKeyPathLen;     /**< @brief Length associated with #NetworkCredentials.pPrivateKey. */
+} OpensslCredentials_t;
+
+/**
+ * @brief Sets up a TLS session on top of a TCP connection using the OpenSSL API.
+ *
+ * @param[in] pNetworkContext Application-defined context (TCP socket and SSL context).
```

Comment:
This is output parameter. 

I'd suggest the following doc for it in the connect API:
```
@param[out] pNetworkContext The output parameter to return the created network context.
```

And the following for the APIs where it is input parameter.
```
@param[in] pNetworkContext The network context created using Openssl_Connect API.
```

The same for Plaintext APIs.